static/app/utils/withIssueTags.tsx

```diff
@@ -27,97 +21,106 @@ const getUsername = ({isManaged, username, email}: User) => {
   return !isManaged && username ? username : email;
 };
 
+type WrappedComponentState = {
+  tags: TagCollection;
+  users: User[];
+  teams: Team[];
+};
 /**
  * HOC for getting tags and many useful issue attributes as 'tags' for use
  * in autocomplete selectors or condition builders.
  */
-function withIssueTags<P extends InjectedTagsProps>(
-  WrappedComponent: React.ComponentType<P>
+function withIssueTags<Props extends WithIssueTagsProps>(
+  WrappedComponent: React.ComponentType<Props>
 ) {
-  class WithIssueTags extends React.Component<
-    Omit<P, keyof InjectedTagsProps> & Partial<InjectedTagsProps>,
-    State
-  > {
-    static displayName = `withIssueTags(${getDisplayName(WrappedComponent)})`;
-
-    constructor(props, context) {
-      super(props, context);
-
-      const tags = assign(
-        {},
+  function ComponentWithTags(props: Omit<Props, keyof WithIssueTagsProps>) {
+    const [state, setState] = React.useState<WrappedComponentState>({
+      tags: assign(
```

Comment:
Should the first arg to assign() be an empty object so that you don't mutate `TagStore.getAllTags()`?

Reply:
100%, nice catch!